### PR TITLE
Add mviewer and mviewerstudio as optional viewer link to #102

### PR DESCRIPTION
**Pull request by pierrejego**
This will add mviewer and mviewerstudio application.

Installation is optional and behind georchestra's security-proxy

### CODE REVIEW

File: roles/apache/templates/mviewerstudio/config.json.j2 [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                },
+                {
+                    "title": "Catalogue de la Région Pays de la Loire",
+                    "url": "https://www.geopal.org/geonetwork/srv/fre/csw",
+                    "baseref": "https://www.geopal.org/geonetwork/srv/eng/catalog.search?node=srv#/metadata/"
+                }
+            ],
+            "wms": [{
+                "title": "Serveur WMS local",
+                "url": "https:/{{ georchestra.fqdn }}/geoserver/rb/wms"
```
Comment (landryb) [inline roles/apache/templates/mviewerstudio/config.json.j2:177]: wont work :)

Comment (pierrejego) [reply roles/apache/templates/mviewerstudio/config.json.j2]: yep good catch, rb workspace removed

**Comment by landryb** [comment]
never used/tried mviewer, not a fan of some things:
- does installing php bring the necessary bits for apache mod_php ? not sure about that..
- data in `/var/www/mviewer/store` ... other georchestra components store data in `/srv/data/<subdir>`
- usually we dont use ports below 1024 for 'alternative' stuff (rather 8xxx), defaults to 85 & 90 look ugly. And that requires root access (yeah i know apache is started as root, but still..)
- what's the point of installing behind the SP ? to restrict access to some roles ? if so, should roles be added/security rules of the SP be amended so that only some role can access the studio ?

**Comment by pierrejego** [comment]
> * does installing php bring the necessary bits for apache mod_php ? not sure about that..
 
yep it does, tested on debian 11 with this playbook branch

> * data in `/var/www/mviewer/store` ... other georchestra components store data in `/srv/data/<subdir>`

mviewer stores its applications configurations files in /var/www/mviewer/apps, /var/www/mviewer/apps/store is only a sub-dir for applications app generated by mviewerstudio
https://mviewerdoc.readthedocs.io/fr/latest/doc_tech/config_practices.html#organisation-des-fichiers-de-carte
Some symlinks could be done to store in /srv/data/mviewer/apps if you link for geOrchestra integration but is it really needed ?

> * usually we dont use ports below 1024 for 'alternative' stuff (rather 8xxx), defaults to 85 & 90 look ugly. And that requires root access (yeah i know apache is started as root, but still..)

Never earded before this was ugly. You can change for any other port in the inventory. What number would you link by default ?  
```
mviewer:
      enabled: true
      port: 85
```
```
   mviewerstudio:
      enabled: true
      port: 90
```

> * what's the point of installing behind the SP ? to restrict access to some roles ? if so, should roles be added/security rules of the SP be amended so that only some role can access the studio ?

There are three different points of beeing behind the security-proxy : 

*  Application generated from mviewerStudio will store user information in configuration files 
```
<dc:creator>pjego</dc:creator>
<dc:publisher>aadam</dc:publisher>
<dc:date>2022-11-24T10:16:20.242Z</dc:date>
```

* When you access directly to an application by url, and when the application use protected OGC services  (from geOrchestra), you will have to login via CAS, otherwise mviewer will ask for multiple basic-authent login/mdp


* And as you said, if you want you can restrict access to some roles from accessing mviewer or mviewerstudio. This is the case for specifis mviewer applications but for me by default no restriction is needed. Some will restricted mviewerstudio access to connected user or admin. But not by default.

**Comment by landryb** [comment]
> > * does installing php bring the necessary bits for apache mod_php ? not sure about that..
> 
> yep it does, tested on debian 11 with this playbook branch
> 
> > * data in `/var/www/mviewer/store` ... other georchestra components store data in `/srv/data/<subdir>`
> 
> mviewer stores its applications configurations files in /var/www/mviewer/apps, /var/www/mviewer/apps/store is only a sub-dir for applications app generated by mviewerstudio https://mviewerdoc.readthedocs.io/fr/latest/doc_tech/config_practices.html#organisation-des-fichiers-de-carte Some symlinks could be done to store in /srv/data/mviewer/apps if you link for geOrchestra integration but is it really needed ?

symlinks are just an ugly workaround. In general, for security reasons/separation of concerns its better for web applications to store their data directories (where the user running the webapp should be able to write) *outside* the webapp source directory, otherwise it needs special scrutiny (eg making sure in the webserver config no other applications or only whitelisted applications could read/write to it, making sure it cant rewrite its own code, etc..) - but given that everything is running under the apache process umbrella as www-data user... oh well, sigh :)

> 
> > * usually we dont use ports below 1024 for 'alternative' stuff (rather 8xxx), defaults to 85 & 90 look ugly. And that requires root access (yeah i know apache is started as root, but still..)
> 
> Never earded before this was ugly. You can change for any other port in the inventory. What number would you link by default ?

usually, most ports below 1024 are reserved, but even if 85/90 aren't, its not a reason to use them as they require you to run things as root, while (say, if running with nginx & php-fpm) one could perfectly run mviewer on a port above 1024 as an unpriviledged user instead of www-data, which is 'dangerous' because shared with all other webapps running in the same web server... but that comes as a default with apache and years of bad security practices :)

given that datafeeder uses 8480 and gn-cloud-searching uses 8580, maybe 8680 and 8780 would make sense ? its not a matter of personal taste, just for consistency with the existing modules...

> > * what's the point of installing behind the SP ? to restrict access to some roles ? if so, should roles be added/security rules of the SP be amended so that only some role can access the studio ?
> 
> There are three different points of beeing behind the security-proxy :
> 
>     * Application generated from mviewerStudio will store user information in configuration files
> 
> 
> ```
> <dc:creator>pjego</dc:creator>
> <dc:publisher>aadam</dc:publisher>
> <dc:date>2022-11-24T10:16:20.242Z</dc:date>
> ```
> 
>     * When you access directly to an application by url, and when the application use protected OGC services  (from geOrchestra), you will have to login via CAS, otherwise mviewer will ask for multiple basic-authent login/mdp
> 
>     * And as you said, if you want you can restrict access to some roles from accessing mviewer or mviewerstudio. This is the case for specifis mviewer applications but for me by default no restriction is needed. Some will restricted mviewerstudio access to connected user or admin. But not by default.

ok, makes sense, thanks for the explanation.

**Pull request merged**